metadata-ingestion/src/datahub/utilities/perf_timer.py

```diff
@@ -6,21 +6,53 @@
 class PerfTimer(AbstractContextManager):
     """
     A context manager that gives easy access to elapsed time for performance measurement.
+
     """
 
-    start_time: Optional[float] = None
-    end_time: Optional[float] = None
+    def __init__(self) -> None:
+        self.start_time: Optional[float] = None
+        self.end_time: Optional[float] = None
+        self._past_active_time: float = 0
+        self.paused: Optional[bool] = None
 
     def start(self) -> None:
+        # TODO
+        # assert (
+        #    self.end_time is None
+        # ), "Can not start a finished timer. Did you accidentally re-use this timer ?"
+
+        if self.end_time is not None:
+            self._past_active_time = self.elapsed_seconds()
+
         self.start_time = time.perf_counter()
         self.end_time = None
+        if self.paused:
+            self.paused = False
+
+    def pause_timer(self) -> "PerfTimer":
```

Comment:
Thoughts on renaming this to just `pause()`. Generally it's clear that the object being paused is a timer

Reply:
makes sense.